
/EkfCalNode:
    ros__parameters:
        debug_log_level: 2
        data_logging_on: true
        body_data_rate: 10.0
        sim_params:
            seed: 0.0
            use_seed: true
            max_time: 10.0
            number_of_runs: 10
            run_number: 0
            stationary_time: 1.0
            truth_type: "spline"
            positions:
                - [00.0, 00.0, 00.0]
                - [10.0, 05.0, 10.0]
                - [20.0, 20.0, 25.0]
                - [35.0, 30.0, 30.0]
            angles:
                - [0.00, 0.00, 0.00]
                - [0.01, 0.01, 0.01]
                - [0.00, 0.00, 0.00]
                - [0.01, 0.00, 0.01]

        filter_params:
            process_noise: [
                1.0e-3, 1.0e-3, 1.0e-3, # Position
                1.0e-4, 1.0e-4, 1.0e-4, # Velocity
                1.0e-5, 1.0e-5, 1.0e-5, # Acceleration
                1.0e-3, 1.0e-3, 1.0e-3, # Angular Position
                1.0e-4, 1.0e-4, 1.0e-4, # Angular Velocity
                1.0e-5, 1.0e-5, 1.0e-5, # Angular Acceleration
            ]

        imu_list:
            - "vn300"

        camera_list:

        tracker_list:

        fiducial_list:

        gps_list:
            - "gps_1"

        imu:
            vn300:
                use_for_prediction: false
                is_extrinsic: false
                is_intrinsic: false
                rate: 400.0
                topic: "/vn300/imu_uncompensated"
                variance: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
                pos_i_in_b: [0.0, 0.0, 0.0]
                ang_i_to_b: [1.0, 0.0, 0.0, 0.0]
                acc_bias: [0.0, 0.0, 0.0]
                omg_bias: [0.0, 0.0, 0.0]
                pos_stability: 1.0e-2
                ang_stability: 1.0e-2
                acc_bias_stability: 1.0e-3
                omg_bias_stability: 1.0e-3
                data_log_rate: 10.0
                sim_params:
                    no_errors: false
                    time_bias_error: 1.0e-3
                    time_skew_error: 1.0e-9
                    time_error: 1.0e-6
                    pos_error: [0.0, 0.0, 0.0]
                    ang_error: [0.0, 0.0, 0.0]
                    acc_error: [1.0e-3, 1.0e-3, 1.0e-3]
                    omg_error: [1.0e-2, 1.0e-2, 1.0e-2]
                    acc_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]
                    omg_bias_error: [1.0e-1, 1.0e-1, 1.0e-1]

        camera:

        tracker:

        gps:
            gps_1:
                rate: 5.0
                topic: "/gps1"
                pos_a_in_b: [0.0, 0.0, 0.0]
                pos_l_in_g: [0.0, 0.0, 0.0]
                ang_l_to_g: 0.0
                variance: [5.0, 5.0, 5.0]
                data_log_rate: 5.0
                quality_limit: 0.1
                sim_params:
                    no_errors: false
                    time_bias: 0.0
                    time_skew: 0.0
                    time_error: 1.0e-9
                    lla_error: [5.0, 5.0, 5.0]
                    pos_a_in_b_err: [1.0e-9, 1.0e-9, 1.0e-9]
                    pos_l_in_g_err: [1.0e-9, 1.0e-9, 1.0e-9]
                    ang_l_to_g_err: 1.0e-9
